xds/internal/xdsclient/clientimpl_dump.go

```diff
@@ -22,27 +22,33 @@ import (
 	v3statuspb "github.com/envoyproxy/go-control-plane/envoy/service/status/v3"
 )
 
-// DumpResources returns the status and contents of all xDS resources.
-func (c *clientImpl) DumpResources() (*v3statuspb.ClientStatusResponse, error) {
+// dumpResources returns the status and contents of all xDS resources.
+func (c *clientImpl) dumpResources() *v3statuspb.ClientConfig {
 	c.authorityMu.Lock()
 	defer c.authorityMu.Unlock()
 
 	var retCfg []*v3statuspb.ClientConfig_GenericXdsConfig
 	for _, a := range c.authorities {
-		cfg, err := a.dumpResources()
-		if err != nil {
-			return nil, err
-		}
-		retCfg = append(retCfg, cfg...)
+		retCfg = append(retCfg, a.dumpResources()...)
 	}
 
-	return &v3statuspb.ClientStatusResponse{
-		Config: []*v3statuspb.ClientConfig{
-			{
-				// TODO: Populate ClientScope. Need to update go-control-plane dependency.
-				Node:              c.config.Node(),
-				GenericXdsConfigs: retCfg,
-			},
-		},
-	}, nil
+	return &v3statuspb.ClientConfig{
+		Node:              c.config.Node(),
+		GenericXdsConfigs: retCfg,
+	}
+}
+
+// DumpResources returns the status and contents of all xDS resources, in a
+// ClientStatusResponse message.
+func DumpResources() *v3statuspb.ClientStatusResponse {
+	clientsMu.Lock()
+	defer clientsMu.Unlock()
+
+	resp := &v3statuspb.ClientStatusResponse{}
+	for key, client := range clients {
```

Comment:
Optional: I don't think this matters, but do we want this to be deterministic? This is related to the new Go heuristic? rule? Doug brought up for the nondeterministic metadata parsing: "Maps are printed in a consistent order, sorted by the values of the keys." I know this isn't really printing it, just building out a response.

Reply:
Mark mentioned in chat that CSDS has no guarantee of ordering, and the deterministic map thing is Go specific, so this is ok.